sbc_switzerland/models/partner.py

```diff
@@ -39,3 +41,13 @@ def write(self, vals):
                     tc.upsert_user(partner, create=False)
 
         return res
+
+    @api.multi
+    def agree_to_child_protection_charter(self):
+        res = super(ResPartner, self).agree_to_child_protection_charter()
+        translation = self.env.ref('partner_compassion.engagement_translation')
+        for partner in self:
+            advocate = partner.advocate_details_id
+            if translation in advocate.engagement_ids:
+                advocate.set_active()
```

Comment:
Yes, I confirm that the partner is activated when he agrees to the child protection charter

Reply:
:+1: